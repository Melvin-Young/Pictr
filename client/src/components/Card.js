import React from 'react';

const Card = ({ src }) => (
	<div className="card">
		<img alt="Flickr" src={src} />
	</div>
);

export default Card;
