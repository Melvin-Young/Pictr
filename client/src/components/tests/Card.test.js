import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Card from '../Card';
import { photoData } from './Fixtures';

Enzyme.configure({ adapter: new Adapter() });

describe('Card', () => {
	it('should render', () => {
		const component = shallow(<Card src={photoData[0].src}/>);
		expect(component).toMatchSnapshot();
	});
});

