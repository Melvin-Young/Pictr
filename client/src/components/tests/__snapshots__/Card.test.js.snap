// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    src="https://farm5.staticflickr.com/4836/31242683977_ece3744565_m.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="Flickr"
        src="https://farm5.staticflickr.com/4836/31242683977_ece3744565_m.jpg"
      />,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Flickr",
        "src": "https://farm5.staticflickr.com/4836/31242683977_ece3744565_m.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Flickr"
          src="https://farm5.staticflickr.com/4836/31242683977_ece3744565_m.jpg"
        />,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Flickr",
          "src": "https://farm5.staticflickr.com/4836/31242683977_ece3744565_m.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
